File: src/types/workspace-attention.ts
export type WorkspaceAttentionStatus = 'idle' | 'running' | 'waiting' | 'error' | 'complete' | string;

export interface WorkspaceAttention {
  workspaceId: string;
  status: WorkspaceAttentionStatus;
  runningCount: number;
  unreadCount: number;
  lastEvent?: string | null;
  lastEventAt?: string | null;
  queuedCount?: number;
}

File: frontend/src-tauri/src/models/workspace_attention.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceAttention {
    pub workspace_id: String,
    pub status: String,
    pub running_count: i64,
    pub unread_count: i64,
    pub last_event: Option<String>,
    pub last_event_at: Option<String>,
}
